include/glow/Support/Error.h

```diff
@@ -437,7 +437,7 @@ class GlowError : protected detail::CheckState<detail::enableCheckingErrors> {
              "Trying to skip state check on an Error that "
              "contains an ErrorValue is a bug because this should only happen "
              "in a constructor and then no ErrorValue should be contained.");
-    } else {
+    } else if (hasErrorValue()) {
```

Comment:
This is here intentionally, Errors should be checked always, not just when there is an ErrorValue. The point of having this to, absent a way to statically check this in cpp, ensure that any time an Error exists it get's checked before being destroyed

Reply:
This makes sense, but I on Windows I have the case when devPromise will make an empty GlowError object, which will be replaced using operator=. That empty object won't be checked, so, when using operator=, the assert will throw the error.

On Linux, this happens using placement new, which will then call the constructor GlowError(GlowError &&other), which skips the checks.
https://github.com/llvm/llvm-project/blob/main/libcxx/include/future#L653
https://github.com/gcc-mirror/gcc/blob/master/libstdc++-v3/include/std/future#L250

Another idea will be to also skip the check in operator=. Is this better? Do you have any other ideas how to differentiate dummy GlowError from the ones needed to be checked?